File: A-Star Pathfinding (Unity)/Node.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Node : IHeapItem<Node>
{
    // Nodes have 2 states: walkable or not walkable
    public bool walkable;
    // The position this node represents in the world
    public Vector3 worldPosition;
    public int gridX, gridY;

    public int gCost;
    public int hCost;
    public int weight;

    public Node parent;
    int heapIndex;

    public Node(bool _walkable, Vector3 _worldPos, int _gridX, int _gridY, int _weight = 0)
    {
        walkable = _walkable;
        worldPosition = _worldPos;
        gridX = _gridX;
        gridY = _gridY;
        weight = _weight;
    }

    public int fCost { get { return gCost + hCost; } }

    public int HeapIndex { 
        get { return heapIndex; } 
        set { heapIndex = value; } 
    }

    public int CompareTo(Node nodeToCompare)
    {
        int compare = fCost.CompareTo(nodeToCompare.fCost);
        if (compare == 0)
        {
            compare = hCost.CompareTo(nodeToCompare.hCost);
        }
        return -compare;
    }
}

File: A-star pathfinding/Node.py
class Node:
    def __init__(self, pos, gridX, gridY):
        self.type = 0       # Determines type of node (0 = walkable, 1 = target, 2 = start, 3 = obstacle, 4 = path, 5 = open, 6 = closed)
        self.worldPosition = pos
        self.gridX = gridX
        self.gridY = gridY
        self.gCost = 0
        self.hCost = 0
        self.parent = None
        self.heapIndex = 0

    def fCost(self):
        return self.gCost + self.hCost

    def get_heapIndex(self):
        return self.heapIndex

    def set_heapIndex(self, value):
        self.heapIndex = value

    def CompareTo(self, node):
        # Returns 1 if cost is lower than compared node, or -1 if cost is higher
        compare = self.fCost() - node.fCost()
        if compare == 0:
            compare = self.hCost - node.hCost

        if compare > 0:
            compare = -1
        elif compare < 0:
            compare = 1

        return compare
